pushers/pusher.go

```diff
@@ -198,7 +198,14 @@ func (p *Pusher) flush(ctx context.Context, payload []byte, flushRespC chan []by
 		log.Printf("Failed to flush logs, err: %v", err)
 		if len(payload) > p.bufferSize {
 			log.Printf("Dropping logs")
-			payload = payload[p.bufferSize/10:] // Drop 1/10 of logs
+			start := p.bufferSize / 10
```

Comment:
This is fine but I think logs are timestamp sorted, instead of dropping with hardcoded value, we can do binary search and drop logs with older than x minute logic

Reply:
Is there a need for binary search here as we are just trying to decrease the number of entries for the payload, we can do a linear search until number of entries are smaller then buffer size and separator is found

Reply:
To make dropping logic more deterministic based on older times I suggested binary search but it requires dynamic buffersize imho

Reply:
dropping based on timestamp would require large code changes, we can do it if it is necessary in a later PR.